### ISSUE TITLE
Loop variable types depend on file-checking order

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ty#4577]
### Summary

Download the playground and run the following commands:

```sh
RAYON_NUM_THREADS=1 ty check --config-file ty.toml check_x.py check_y.py
RAYON_NUM_THREADS=1 ty check --config-file ty.toml check_y.py check_x.py
```

| Command-line order | `x` in `check_x.py` | `y` in `check_y.py` | Observations |
| --- | --- | --- | --- |
| `check_x.py check_y.py` | `Never` | `Literal[True]` | 12 of 12 runs |
| `check_y.py check_x.py` | `Literal["large"]` | `Never` | 12 of 12 runs |

The checked file set and every file's contents are identical. This changes the actual revealed types, which could result in diagnostic changes.

`check_x.py`:

```py
from typing import reveal_type
from m import x

reveal_type(x)

```

`check_y.py`:

```py
from typing import reveal_type
from m import y

reveal_type(y)

```

`m.py`:

```py
from c import Large, a as orig, b as orig2, what

a = orig
b = orig2
while True:
    x = a
    y = b
    if y:
        pass
    else:
        a = Large
    if x:
        pass
    else:
        b = True
    if what:
        break

```

`c.pyi`:

```py
from typing import Literal, Never

a: Never
b: Never
what: bool
Large: Literal["large"]

```

`ty.json`:

```json
{
  "environment": {
    "python-version": "3.13"
  }
}

```

[Playground](https://play.ty.dev/fb2aac41-41e8-4923-a0a6-01f6ca560c90)
